src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6262,7 +6245,7 @@ private enum ParseTypeMode
                     while (this.IsPossibleRankAndDimensionSpecifier())
                     {
                         bool unused;
-                        var rank = this.ParseArrayRankSpecifier(mode == ParseTypeMode.ArrayCreation, expectSizes, allowQuestionToken: true, out unused);
+                        var rank = this.ParseArrayRankSpecifier(mode == ParseTypeMode.ArrayCreation, expectSizes, questionTokenModeOpt: mode, out unused);
                         ranks.Add(rank);
```

Comment:
This appears to apply to all levels of a multidimensional array. Shouldn't it apply only to the last?  That is, shouldn't we allow `var z = e is int[]?[] ? x : y;` (existing code)

Reply:
We allow `e is int[]?[] ? x : y` and `e is int[]?[]? ? x : y`. The issue was we were incorrectly parsing the first case by treating the `?` from the conditional as a nullable qualifier on the array type.